src/Voron/Data/BTrees/TreeMutableState.cs

```diff
@@ -8,120 +8,69 @@ namespace Voron.Data.BTrees
 {
     public sealed unsafe class TreeMutableState
     {
-        private readonly LowLevelTransaction _tx;
+        private TreeRootHeader _header;
+        private bool _stateIsModified;
 
-        public RootObjectType RootObjectType;
-        public long RootPageNumber;
-        public TreeFlags Flags;
+        public ref readonly TreeRootHeader Header => ref _header;
```

Comment:
It is a `readonly` reference. Most of the time we don't need to modify.

Reply:
We dont really win anything with the properties and force the JIT to have to reason about the access references. Having the ability to access the values from a reference makes the semantics explicit. First, the constraint of the `readonly ref` is enforced by the language and the JIT can use that knowledge to optimize (that is not true when using properties). Second, tracking places where modifications happen is as easy as do a reverse reference search over `.Modify()`. 

After having done this change and see what happens when we apply this methodology I would argue we should use more of this pattern, and get rid as much as possible of the track modification when executing property setters. Being explicit allow us to avoid common errors.